### PR TITLE
jellyseerr: init at 1.2.1

### LINKED ISSUE #176432: Package request: Jellyseerr
**Project description**

Fork of Overseerr for Jellyfin support. Overseerr also isn't packaged. They're comparable to Ombi which is.

**Metadata**

* Homepage / source: https://github.com/Fallenbagel/jellyseerr
* License: MIT
* Platforms: Linux at least - there's a Docker container and an AUR package

### PR DESCRIPTION
**Pull request by tboerger**
###### Description of changes

Initialize new package for jellyseer. Fixes https://github.com/NixOS/nixpkgs/issues/176432

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Generally this package is building fine, except that it for any reason is missing the sqlite3 NodeJS dependency. Maybe somebody who already packaged NodeJS applications got an idea what is happening here?

```
❯ /nix/store/xf7lr35h4cilfvkshs24gfv9l60phcxf-jellyseerr-1.2.1/bin/jellyseerr
/nix/store/nnlqlihcnvjdd66jjlmr9dpcf1813snm-jellyseerr-modules-1.2.1/deps/jellyseerr/node_modules/typeorm/driver/sqlite/SqliteDriver.js:130
            throw new DriverPackageNotInstalledError_1.DriverPackageNotInstalledError("SQLite", "sqlite3");
            ^

DriverPackageNotInstalledError: SQLite package has not been found installed. Try to install it: npm install sqlite3 --save
    at SqliteDriver.loadDependencies (/nix/store/nnlqlihcnvjdd66jjlmr9dpcf1813snm-jellyseerr-modules-1.2.1/deps/jellyseerr/node_modules/typeorm/driver/sqlite/SqliteDriver.js:130:19)
    at new SqliteDriver (/nix/store/nnlqlihcnvjdd66jjlmr9dpcf1813snm-jellyseerr-modules-1.2.1/deps/jellyseerr/node_modules/typeorm/driver/sqlite/SqliteDriver.js:29:14)
    at DriverFactory.create (/nix/store/nnlqlihcnvjdd66jjlmr9dpcf1813snm-jellyseerr-modules-1.2.1/deps/jellyseerr/node_modules/typeorm/driver/DriverFactory.js:44:24)
    at new DataSource (/nix/store/nnlqlihcnvjdd66jjlmr9dpcf1813snm-jellyseerr-modules-1.2.1/deps/jellyseerr/node_modules/typeorm/data-source/DataSource.js:49:59)
    at Object.<anonymous> (/nix/store/xf7lr35h4cilfvkshs24gfv9l60phcxf-jellyseerr-1.2.1/libexec/jellyseerr/deps/jellyseerr/dist/datasource.js:32:20)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)

Node.js v18.12.1
```

### CODE REVIEW

**Comment by nrabulinski** [comment]
[https://github.com/NixOS/nixpkgs/blob/d44a146cc1c6cf91547f4788610a1ded3c4e45d5/pkgs/servers/web-apps/hedgedoc/default.nix](The hedgedoc package) also depends on `sqlite` and what they do is set up `buildPhase` to build the required native dependencies from source instead of fetching vendored ones. That could be a good start

**Comment by tboerger** [comment]
> [The hedgedoc package](https://github.com/NixOS/nixpkgs/blob/d44a146cc1c6cf91547f4788610a1ded3c4e45d5/pkgs/servers/web-apps/hedgedoc/default.nix) also depends on `sqlite` and what they do is set up `buildPhase` to build the required native dependencies from source instead of fetching vendored ones. That could be a good start

I have seen that, but they are also copying the node_modules directory around. Is that really the way to go?

**Comment by tboerger** [comment]
I don't get it working correctly and lacking time to continue on that.

File: pkgs/servers/jellyseerr/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  yarnNix = ./yarn.nix;
+
+  doDist = false;
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+    sqlite
+  ];
+
```
Comment (camillemndn) [inline pkgs/servers/jellyseerr/default.nix:33]:
```

  pkgConfig.sqlite3.postInstall = ''
    export CPPFLAGS="-I${nodejs}/include/node"
    ${nodePackages.node-pre-gyp}/bin/node-pre-gyp install --prefer-offline --build-from-source --python=${python3}/bin/python --nodedir=${nodejs}/include/node
  '';

  pkgConfig.bcrypt.postInstall = ''
    export CPPFLAGS="-I${nodejs}/include/node"
    ${nodePackages.node-pre-gyp}/bin/node-pre-gyp install --prefer-offline --build-from-source --python=${python3}/bin/python --nodedir=${nodejs}/include/node
  '';
```

File: pkgs/servers/jellyseerr/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, mkYarnPackage
+, nodejs
+, sqlite
+, pkgs
```
Comment (camillemndn) [inline pkgs/servers/jellyseerr/default.nix:8]:
```
, python3
```

File: pkgs/servers/jellyseerr/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
+  yarnNix = ./yarn.nix;
+
+  doDist = false;
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+    sqlite
```
Comment (camillemndn) [inline pkgs/servers/jellyseerr/default.nix:31]:
```
    python3
```

**Review by camillemndn — COMMENTED** [review]
Hi, I think I found a fix for your problem with sqlite3, but then I had the same with bcrypt and used the same fix. Tell me if it works for you :)

**Pull request closed without merging**